### ISSUE TITLE
iPhone X selectionIndicatorImage

### ISSUE THREAD

**Issue by SwiftRabbit** [Ramotion/animated-tab-bar#191]
Having an issue trying to set the selectionIndicator Image
I create an image with a black background and this is what I get:

<img width="410" alt="capture d ecran 2017-09-13 a 11 26 35" src="https://user-images.githubusercontent.com/6453814/30387103-8ad91982-9879-11e7-810a-78b7e8d71d5e.png">

**Comment by albertml** [comment]
did you find a solution for this?

**Comment by SwiftRabbit** [comment]
No, still pending :(  (Found another bug making a vertical transition though)

**Comment by albertml** [comment]
Where did you test? In a real device or a simulator?

**Comment by casperzandbergenyaacomm** [comment]
I have the same problem but I got a bit closer using:

> .resizableImage(withCapInsets: UIEdgeInsets(top: 1, left: 0, bottom: 0, right: 0))

This makes the positioning correct but adds padding around it.. :/

**Comment by premanandchandrasekar** [comment]
@casperzandbergenyaacomm solution works for me.

**Comment by tacovollmer** [comment]
It appears `capsInsets` should not match `UIEdgeInsets.zero` for the positioning to be correct on iPhoneX. So using `CGFloat.leastNonzeroMagnitude` for any of the sides also works. e.g.:

`.resizableImage(withCapInsets: UIEdgeInsets(top: .leastNonzeroMagnitude, left: 0, bottom: 0, right: 0))`

**Comment by tacovollmer** [comment]
I haven't been able to fill to the bottom @SureshChary. I'm interested to know if anyone did manage.

**Comment by Amzd** [comment]
@SureshChary I have the same as you and it works for me, are you sure the tabHeight is tabBar.frame.height? 

My code for the bar in the image.

```
let numberOfItems = CGFloat(tabBar.items!.count)
let tabBarItemSize = CGSize(width: tabBar.frame.width/numberOfItems, height: tabBar.frame.height)
let color = tabSelectedColor
        
tabBar.selectionIndicatorImage = UIImage
    .imageWithColor(color: color, size: tabBarItemSize)
    .resizableImage(withCapInsets: UIEdgeInsets(top: 0, left: 0, bottom: 0, right: 0))
```

```
extension UIImage {
    
    class func imageWithColor(color: UIColor, size: CGSize) -> UIImage {
        let rect: CGRect = CGRect(x: 0.0, y: 0.0, width: Double(size.width), height: Double(size.height))
        UIGraphicsBeginImageContextWithOptions(size, false, 0)
        color.setFill()
        UIRectFill(rect)
        let image: UIImage = UIGraphicsGetImageFromCurrentImageContext()!
        UIGraphicsEndImageContext()
        return image
    }
}
```


![image](https://user-images.githubusercontent.com/17001151/37965633-037767b4-31c6-11e8-831a-e34002177355.png)


**Note:** I do not use this library

**Comment by casperzandbergenyaacomm** [comment]
@SureshChary I have the same as you and it works for me, are you sure the tabHeight is tabBar.frame.height? 

My code for the bar in the image.

```
let numberOfItems = CGFloat(tabBar.items!.count)
let tabBarItemSize = CGSize(width: tabBar.frame.width/numberOfItems, height: tabBar.frame.height)
let color = tabSelectedColor
        
tabBar.selectionIndicatorImage = UIImage
    .imageWithColor(color: color, size: tabBarItemSize)
    .resizableImage(withCapInsets: UIEdgeInsets(top: 0, left: 0, bottom: 0, right: 0))
```

```
extension UIImage {
    
    class func imageWithColor(color: UIColor, size: CGSize) -> UIImage {
        let rect: CGRect = CGRect(x: 0.0, y: 0.0, width: Double(size.width), height: Double(size.height))
        UIGraphicsBeginImageContextWithOptions(size, false, 0)
        color.setFill()
        UIRectFill(rect)
        let image: UIImage = UIGraphicsGetImageFromCurrentImageContext()!
        UIGraphicsEndImageContext()
        return image
    }
}
```


![image](https://user-images.githubusercontent.com/17001151/37965633-037767b4-31c6-11e8-831a-e34002177355.png)


**Note:** I do not use this library

**Comment by SureshChary** [comment]
Yes, let tabHeight = tabBar.frame.height

I tried your above code. But again it is failing in iPhone X 11.2. any idea?
Here is code:
`let tabWidth = (tabBar.frame.width/CGFloat(tabBar.items!.count))
        let tabHeight = tabBar.frame.height
        self.tabBar.selectionIndicatorImage = imageWithColor(color: UIColor.white, size: CGSize(width: tabWidth, height: tabHeight)).resizableImage(withCapInsets: UIEdgeInsets(top: 0, left: 0, bottom: 0, right: 0))`

<img width="633" alt="screen shot 2018-03-28 at 12 01 20" src="https://user-images.githubusercontent.com/8349347/38013856-49df78fe-3284-11e8-8b65-47caa5859050.png">

**Comment by casperzandbergenyaacomm** [comment]
@SureshChary You need to call that code in the viewDidLayoutSubviews (after the height sizing for iPhone X)

**Comment by SureshChary** [comment]
Perfect!! Worked and thanks @casperzandbergenyaacomm 

I feel iPhone X Tab UI is ugly bcoz of height-83. is it Ok for Apple?

<img width="639" alt="screen shot 2018-03-28 at 15 08 40" src="https://user-images.githubusercontent.com/8349347/38021412-631e6f26-329a-11e8-8e84-31abc6b84296.png">

**Comment by KoukuntlaAditya** [comment]
I faced the same issue and was able to fix it by adding an extra rect in the imageWithColor extension,

<img width="707" alt="screen shot 2018-10-16 at 7 52 16 pm" src="https://user-images.githubusercontent.com/17068936/47055446-0d5b3b00-d17d-11e8-8099-d8de39ca49cc.png">

This is how it looked after the fix.
<img width="82" alt="screen shot 2018-10-16 at 7 53 10 pm" src="https://user-images.githubusercontent.com/17068936/47055474-29f77300-d17d-11e8-9628-b94548bd3809.png">

**Comment by casperzandbergenyaacomm** [comment]
I would actually advise against using a background color on your tab bar buttons nowadays, they look off on the current generation of iPhones. A selected icon and unselected icon is a better way to show wether the tab is active.
